pkg/bgpv1/types/bgp.go

```diff
@@ -44,6 +44,8 @@ type Path struct {
 type NeighborRequest struct {
 	Neighbor *v2alpha1api.CiliumBGPNeighbor
 	VR       *v2alpha1api.CiliumBGPVirtualRouter
+	// TCPPassword is the "AuthSecret" in the Neighbor, fetched from a secret
```

Comment:
nit: Are there any other supported password types? If not, consider omitting `TCP` throughout to simplify the naming.

Reply:
I've dropped it, it's easy enough to change it later if needed (the other types would be other than "MD5", which we could add a key in the secret for, rather than other auth methods, the password is a bit of a misnomer anyway -- it's actually a passphrase / cipher key, but that's what it is called commonly and in the gobgp API).